packages/frontend/src/notebook/notebook_editor.tsx

```diff
@@ -249,60 +247,74 @@ export function NotebookEditor<T>(props: {
                 </div>
             </Show>
             <ul class="notebook-cells">
-                <For each={props.notebook.cells}>
-                    {(cell, i) => {
+                <For each={props.notebook.cellOrder}>
+                    {(cellId, i) => {
+                        const cell = props.notebook.cellContents[cellId];
+                        if (!cell) {
+                            throw `Failed to find contents for cell ${cellId}`;
+                        }
```

Comment:
I agree that in this key spot, it's better to throw an error than silently ignore the missing cell, because this is an important invariant that should never fail. In fact, using `tiny-invariant`, I like to write these checks as

```typescript
invariant(cell, () => `Failed to find contents for cell ${cellId}`);
```

which saves a line or two of boilerplate.